executor/runtime/docker/docker.go

```diff
@@ -1427,6 +1436,98 @@ func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, c runtimeTypes.
 	}
 }
 
+// startOtherUserContainers launches the other user containers, only looking
+// any container objects in the pod *after* the first one, converting the
+// v1.Container spec into something docker can understand, and then
+// running that container.
+func (r *DockerRuntime) startOtherUserContainers(ctx context.Context, pod *v1.Pod, mainContainerID string) error {
+	if pod == nil || pod.Spec.Containers == nil {
+		return nil
+	}
+	otherUserContainers := pod.Spec.Containers[1:]
+	if len(otherUserContainers) == 0 {
+		return nil
+	}
+	l := log.WithField("taskID", r.c.TaskID())
+	l.Infof("Starting up %d other user containers", len(otherUserContainers))
+	// Currently the order here is hard-coded to be whatever order it is in the podspec
+	// with no health or startup probes
+	for i, v1Container := range otherUserContainers {
+		l.Debugf("Starting other user container %d/%d: %s", i+1, len(otherUserContainers), v1Container.Name)
+		err := pullWithRetries(ctx, r.metrics, r.client, v1Container.Image, doDockerPull)
```

Comment:
I think it's a better idea to pull all containers before we start them. Pulling here might cause unexpectedly long delays in startup.

Reply:
Hmm, yeah - after thinking about it a bit, that seems reasonable.  Starting shouldn't take long (since it's just launching tini at this point), so pulling images is going to be the long pole here.  The image pull seems like the most likely candidate for things to go wrong, so might as well fail up front before we start even one container.

Reply:
Hmm. I'll see if I can parallelize this without loss of logging fidelity.